techlibs/intel_alm/common/bram_m10k.txt

```diff
@@ -0,0 +1,33 @@
+bram __MISTRAL_M10K_SDP
+    init   0   # TODO: Re-enable when I figure out how BRAM init works
+    abits 13   @D8192x1
+    dbits  1   @D8192x1
+    abits 12   @D4096x2
+    dbits  2   @D4096x2
+    abits 11   @D2048x4 @D2048x5
+    dbits  4   @D2048x4
+    dbits  5   @D2048x5
+    abits 10   @D1024x8 @D1024x10
+    dbits  8   @D1024x8
+    dbits 10   @D1024x10
+    abits  9   @D512x16 @D512x20
+    dbits 16   @D512x16
+    dbits 20   @D512x20
+    abits  8   @D256x32 @D256x40
+    dbits 32   @D256x32
+    dbits 40   @D256x40
+    groups 2
+    ports  1 1
+    wrmode 1 0
+    # read enable; write enable + byte enables (only for multiples of 8)
+    enable 1 1
```

Comment:
This describes a blockram with only a single write enable (not byte enables) — is that correct? You can look at Xilinx `xc6s_brams.txt` for an example with byte enables.

Reply:
I'm trying not to run before I can walk; the `altsyncram` primitive is an utter mess and wacky definitions of what a byte is depending on how you configure the memory. For now, a single enable will do.